### PR TITLE
Fix Kinesis source bug

### PR DESCRIPTION
**Pull request by JLDLaughlin**
Fixes https://github.com/MaterializeInc/materialize/issues/2265

### The Embarrassing Problem
`GetRecords` returns a response that contains `millis_behind_latest` and `records`. `millis_behind_latest` indicates the number of milliseconds the get request/consumer is behind the tip of the stream. In the prior implementation of this code, I was mistakenly breaking on `millis_behind_latest == 0` **before** grabbing any records off of the response -- effectively dropping those records on the floor. 

tldr; I did this to myself.

Tested this locally by continuously pushing test records to my AWS Kinesis stream and having them show up in my materialized view of the Kinesis source. More testing to come: https://github.com/MaterializeInc/materialize/issues/2196, https://github.com/MaterializeInc/materialize/issues/2197, https://github.com/MaterializeInc/materialize/issues/2202.

### CODE REVIEW

File: src/dataflow/source/kinesis.rs [hunk 1]
```diff
@@ -171,6 +158,18 @@ where
                     output.session(&cap).give((data, None))
                 }
 
+                if let Some(millis) = get_records_output.millis_behind_latest {
```
Comment (benesch) [inline src/dataflow/source/kinesis.rs]: Definitely don't go back through CI for this, but I think `if let Some(0) = get_records_output.millis_behind_latest` would work here!
